### PR TITLE
🕑 Support playout delay extension

### PR DESCRIPTION
**Pull request by danstiner**
### TODO
- [x] Test on a production server
- [x] Do not send playout delay on every packet, once the client has received one playout delay it will keep using that delay. Probably sending on the first N packets would be enough.

### Summary
This PR enables support for the experimental playout delay extension described here: https://webrtc.googlesource.com/src/+/refs/heads/main/docs/native-code/rtp-hdrext/playout-delay

The intention is to test playout delay as a workaround for the issue discussed in #101, at least until we can look deeper at changing Chrome to correctly determine a playout delay.

### Requirements
- Install patched janus-gateway that supports the playout-delay extension (https://github.com/danstiner/janus-gateway/tree/playout-delay-extension)
- Build&install this plugin with meson option `janus_playout_delay_support=true`
- Launch janus-gateway with the environment variables FTL_PLAYOUT_DELAY_MIN_MS and FTL_PLAYOUT_DELAY_MAX_MS set

### Other Notes
Suggest trying a minimum in the range of 200-400 and a maximum of at least 2000, but no more than 10,000.

Expected outcome is that rendering will be delayed by the min value set. You can try setting a large minimum to see the effect (e.g. 5000), or look at `chrome://webrtc-internals/` like the following image.

This shows a stream with a min playout of 200ms. The delay jumps up to 250ms at one point, but never drops below the configured 200ms min delay.
image

### CODE REVIEW

**Comment by danstiner** [comment]
Reserving comment to discuss testing approach

```

http://linux-ip.net/gl/tc-filters/tc-filters-node3.html
modprobe ifb
ip link set ifb0 up
tc qdisc add dev ifb0 root sfq
tc qdisc add dev eth1 handle ffff: ingress
tc filter add dev eth1 parent ffff: u32 \
        match u32 0 0 \
        action mirred egress redirect dev ifb0


https://www.excentis.com/blog/use-linux-traffic-control-impairment-node-test-environment-part-2

sudo tc qdisc del dev ifb0 root
sudo tc qdisc del dev eth1 root

#sudo tc qdisc change dev ifb0 root netem loss 0.3% 25%

sudo tc qdisc add dev ifb0 root tbf rate 6mbit burst 320kbit latency 400ms
sudo tc qdisc add dev eth1 root netem delay 100ms 10ms 25%
```

File: src/Configuration.cpp [hunk 1]
```diff
@@ -120,25 +122,43 @@ void Configuration::Load()
...
+        nackLostPackets = std::stoul(varVal);
+    }
+
+    // FTL_PLAYOUT_DELAY_MIN_MS and FTL_PLAYOUT_DELAY_MAX_MS -> PlayoutDelay
+    {
+        char* minVal = std::getenv("FTL_PLAYOUT_DELAY_MIN_MS");
+        char* maxVal = std::getenv("FTL_PLAYOUT_DELAY_MAX_MS");
+
+        if (minVal || maxVal) {
+            if (!PLAYOUT_DELAY_SUPPORT) {
```
Comment (haydenmc) [inline src/Configuration.cpp:152]: Braces on new line pls [:>](https://theprofoundprogrammer.com/post/26955601996/text-braces-go-on-the-next-line-you-tremendous)

File: src/Rtp/RtpPacket.cpp [hunk 2]
```diff
@@ -10,17 +10,17 @@
 #include <netinet/in.h>
 
 #pragma region Static utility methods
-const RtpHeader* RtpPacket::GetRtpHeader(const std::vector<std::byte>& rtpPacket)
+const RtpHeader* RtpPacket::GetRtpHeader(std::span<const std::byte> rtpPacket)
```
Comment (haydenmc) [inline src/Rtp/RtpPacket.cpp:13]: 👌👌👌

**Review by haydenmc — APPROVED** [review]
If GitHub's CR tools were good, I'd mark this "approved with suggestions" 😁

File: src/Configuration.cpp (hunk 1 above)
Comment (danstiner) [reply src/Configuration.cpp]: image

thx for the catch :)

**Pull request merged**